tedana/selection/tedica.py

```diff
@@ -273,9 +273,17 @@ def kundu_selection_v2(comptable, n_echos, n_vols):
     # NOTE: Would an elbow from all Kappa values *ever* be lower than one from
     # a subset of lower (i.e., nonsignificant) values?
     kappa_elbow = np.nanmin((kappas_all_elbow, kappas_nonsig_elbow))
+    if kappa_elbow == kappas_all_elbow:
+        LGR.info(f"Kappa elbow is {kappa_elbow} and calculated based on all components")
+    else:
+        LGR.info(
+            f"Kappa elbow is {kappa_elbow} and calculated based on {kappas_nonsig.size}"
+            f" components with non-significant kappa fits"
+        )
```

Comment:
```suggestion
        LGR.info(
            f"Kappa elbow is {kappa_elbow} and calculated based on {kappas_nonsig.size} "
            "components with non-significant kappa fits"
        )
```

No need for the f-string in the second one. Also, I think that trailing spaces are easier to read that leading ones.

Reply:
On a related note, is this actually true? We're using the minimum between the two values. While it's hard to imagine a situation where the elbow from the nonsignificant Kappas would be higher than the elbow from all Kappas, there are more straightforward ways to write the elbow selection if we did expect this to always be the case.

Reply:
nonsignificant refers to a the fit to the kappa model, not necessarily the magnitude of the magnitude of the kappa values. My assumption is that either could be higher, but I don't actually know because we never logged this info... until now!

Reply:
Given that the degrees of freedom are the same across components, wouldn't nonsignificant kappa values _have_ to be smaller than significant ones?

Reply:
You're fitting a line with 3-5 data points. The magnitude of the fit could be higher but the residual could be even larger. Therefore, you can have a higher kappa magnitude that isn't significant. I'd have to double-check the code to confirm this, but that's my memory.

Reply:
I relooked at the code & you're correct. The nonsig threshold is made purely based on the raw kappa value. That's odd on many levels (which is probably why I was mis-remembering it to be more logical)
https://github.com/handwerkerd/tedana/blob/e6f714cadaf8eecaf2e43e879d362e876523c86a/tedana/selection/tedica.py#L261

If the kappa values roughly follow the assumed elbow shape, I agree that you can't have nonsig kappa elbow ever higher than the kappa elbow using all the components. If the distribution of kappa values is weirder, I think it can happen. I've been playing with artificial values to as input to to getelbow and I can't create an immediate demo, so maybe it's not possible in practice?

In all, there are the following scenarios:
1. There aren't enough nonsig kappa values. This code will tell the user that and use the elbow from all kappas
2. There are enough nonsig kappa values and the user will see that the nonsig kappa elbow is used.
3. If there's a questionable case where there are enough nonsig values, but the kappa elbow is lower, then the output would show the full kappa elbow was used, but there isn't a message about too few nonsig kappas.

At minimum that means, this code will let us know what is currently happening.

This all said, this is making me want to increase the nonsig threshold. If the nonsig threshold is nearly always used when it exists, then it should include enough data points to represent an actual elbow. 5? Thoughts?

Reply:
The kappa values should be (averaged) F-statistics though, rather than parameter estimates, so we evaluate their significance based on a single threshold. 

I think that the degrees of freedom probably vary _slightly_ across components, since we're averaging across voxels using component-specific weights, but I think that's something we'd need to add to our calculated metrics and then account for when we call `getfbounds` to get our F-statistic significance threshold.

Reply:
Ah, sorry, my response was to your earlier comment. I guess you posted while I was writing my own.

Reply:
I keep flipping on this. I'm again leaning towards keeping this PR as is and calculating an elbow with just two components. This is wrong, but it's shouldn't change existing results. This part of the code deserves serious rethinking (hopefully after decision tree modularization) but an algorithmic change probably shouldn't be mixed up with this minor bug fix. Perhaps move this discussion to a new issue so we remember to revisit?

Reply:
I just opened #810 to separate this discussion from the specific change proposed in the PR.